src/chunk.c

```diff
@@ -2766,18 +2766,29 @@ init_scan_by_compressed_chunk_id(ScanIterator *iterator, int32 compressed_chunk_
 }
 
 bool
-ts_chunk_contains_compressed_data(Chunk *chunk)
+ts_chunk_contains_compressed_data(Chunk *chunk, Chunk **parent)
 {
 	ScanIterator iterator = ts_scan_iterator_create(CHUNK, AccessShareLock, CurrentMemoryContext);
-	bool found = false;
+	Datum parent_id = 0;
 
 	init_scan_by_compressed_chunk_id(&iterator, chunk->fd.id);
 	ts_scanner_foreach(&iterator)
 	{
-		Assert(!found);
-		found = true;
+		TupleInfo *ti = ts_scan_iterator_tuple_info(&iterator);
+		Assert(!parent_id);
+		bool isnull;
+		parent_id = heap_getattr(ti->tuple, Anum_chunk_id, ti->desc, &isnull);
 	}
-	return found;
+
+	if (parent_id)
```

Comment:
I would avoid these kinds of checks on integers, since it is ambiguous what is a true expression here, especially for a Datum type which can also have NULL values via the separate` isnull` check. I would just move the `isnull` out of the loop and use that here instead.